pkg/api/2019-04-30/types.go

```diff
@@ -138,6 +141,12 @@ type AgentPoolProfile struct {
 	Role *AgentPoolProfileRole `json:"role,omitempty"`
 }
 
+// MonitorProfile configuration for Azure Monitor log analytics.
+type MonitorProfile struct {
+	WorkspaceID  *string `json:"workspaceId,omitempty"`
+	WorkspaceKey *string `json:"workspaceKey,omitempty"`
```

Comment:
I thought WorkspaceKey wasn't going into the external API?

Reply:
I think it's part of the external API, just not the admin API nor the az client.

Reply:
`WorkspaceKey` shouldn't be in the public API model because RP will query it.